### PR TITLE
Import GLFW into //third_party/glfw

### PR DESCRIPTION
**Pull request by abarth**
This patch includes GLFW 3.1.2.

### CODE REVIEW

**Comment by abarth** [comment]
@collinjackson

**Comment by collinjackson** [comment]
lgtm

File: third_party/glfw/BUILD.gn [hunk 1]
```diff
@@ -0,0 +1,51 @@
+# Copyright 2016 The Chromium Authors. All rights reserved.
+# Use of this source code is governed by a BSD-style license that can be
+# found in the LICENSE file.
```
Comment (Hixie) [inline third_party/glfw/BUILD.gn]: what LICENSE file?

Comment (abarth) [inline third_party/glfw/BUILD.gn]: The one in the root of the repo.  There isn't a LICENSE file in this directory.  The license is contained in a file called COPYING.txt, matching the upstream glfw repository.  This is similar to what we and Chromium do for libpng, etc.

**Comment by chinmaygarde** [comment]
OMG, whats this for?

**Comment by collinjackson** [comment]
making Linux builds draw to the screen

**Pull request merged**